### PR TITLE
offline-plugin

### PR DESCRIPTION
**Pull request by mansour-ahmed**
This PR does the following:
- include build files in gitignore
- http server
- offline plugin support
- generic functions that listen to offline, online events

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,11 +2,12 @@
   "name": "react-redux-skeleton",
   "version": "1.0.0",
   "description": "",
-  "main": "index.js",
+  "main": "index.js",
```
Comment (romandunets) [inline package.json]: Extra space

File: src/index.js [hunk 2]
```diff
@@ -1,16 +1,62 @@
...
-import { Router, browserHistory } from 'react-router';
+import {render} from 'react-dom';
+import {Provider} from 'react-redux';
+import {Router, browserHistory} from 'react-router';
 
 import configureStore from './store/configureStore.js';
 import routes from './routes.js';
 
 const store = configureStore();
+import * as OfflinePluginRuntime from 'offline-plugin/runtime';
```
Comment (romandunets) [inline src/index.js]: Maybe import can be moved on top before `const store = configureStore();`

File: webpack.config.js [hunk 3]
```diff
@@ -2,12 +2,11 @@ const webpack = require('webpack');
...
 const buildPath = path.resolve(__dirname, 'build');
 const TransferWebpackPlugin = require('transfer-webpack-plugin');
+const OfflinePlugin = require('offline-plugin');
 
 const config = {
   entry: [
-    'webpack/hot/dev-server',
-    'webpack/hot/only-dev-server',
-    path.join(__dirname, '/src/index.js')
+    'webpack/hot/dev-server', 'webpack/hot/only-dev-server', path.join(__dirname, '/src/index.js')
```
Comment (romandunets) [inline webpack.config.js]: It would be nice if each of these could be on a new line

File: src/index.js [hunk 4]
```diff
@@ -1,16 +1,62 @@
...
+  console.log('closeOfflineReady');
+}
 
 render(
   <Provider store={store}>
-    <Router history={browserHistory} routes={routes(store)} />
-  </Provider>,
-  document.getElementById('app')
-);
+  <Router history={browserHistory} routes={routes(store)}/>
```
Comment (romandunets) [inline src/index.js]: Please check aligning here

File: src/index.js [hunk 5]
```diff
@@ -1,16 +1,62 @@
...
+    OfflinePluginRuntime.applyUpdate();
+  },
+  onUpdated: function () {
+    window
+      .location
+      .reload();
+  }
+});
+
+var rotation = 0;
```
Comment (romandunets) [inline src/index.js]: I do not see the point of this var

**Review by romandunets — COMMENTED** [review]
Good start. Please check formatting issues

**Comment by mansour-ahmed** [comment]
@romandunets PR updated.

**Review by romandunets — APPROVED** [review]

**Pull request merged**